### PR TITLE
py-dask-glm: Added document build

### PR DESCRIPTION
**Pull request by a-saitoh-fj**
py-dask-glm: Added document build

Note that py-s3fs (#22260) is required to build the document.

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-dask-glm/package.py [hunk 1]
```diff
@@ -15,10 +15,33 @@ class PyDaskGlm(PythonPackage):
...
+    depends_on('py-numpydoc', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx-rtd-theme', type=('build', 'run'), when='+docs')
+    depends_on('pandoc', type='build', when='+docs')
+    depends_on('py-pip', type=('build', 'run'), when='+docs')
+    depends_on('py-s3fs', type=('build', 'run'), when='+docs')
+    depends_on('py-matplotlib', type=('build', 'run'), when='+docs')
+    depends_on('llvm@:10.0.1~flang', when='+docs')
+    depends_on('cairo+X+ft+fc+pdf+gobject', type=('build', 'run'), when='+docs')
+    depends_on('harfbuzz+graphite2', type=('build', 'run'), when='+docs')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dask-glm/package.py]: All doc deps should be type=build, the deps aren't needed at run-time.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Comment by a-saitoh-fj** [comment]
py-dask-glm
Push again for testing with git.

In the same way as #22260, I did the following:
I deleted the branch and recreated it with the same name, and, I pushed 'git push --force'.

**Comment by a-saitoh-fj** [comment]
@adamjstewart 

py-dask-glm
Fixed the pointed out OSS dependency setting to type=build.

Sorry for the late response.

File: var/spack/repos/builtin/packages/py-dask-glm/package.py [hunk 2]
```diff
@@ -15,10 +15,33 @@ class PyDaskGlm(PythonPackage):
...
+    depends_on('py-scikit-learn@0.18:0.21.999', type=('build', 'run'), when='+docs')
+    depends_on('py-jupyter', type=('build', 'run'), when='+docs')
+    depends_on('py-nbsphinx', type=('build', 'run'), when='+docs')
+    depends_on('py-notebook', type=('build', 'run'), when='+docs')
+    depends_on('py-numpydoc', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx', type=('build', 'run'), when='+docs')
+    depends_on('py-sphinx-rtd-theme', type=('build', 'run'), when='+docs')
+    depends_on('pandoc', type='build', when='+docs')
+    depends_on('py-pip', type=('build', 'run'), when='+docs')
+    depends_on('py-s3fs', type=('build', 'run'), when='+docs')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dask-glm/package.py]: The rest of these doc deps should be type=build, right?

**Comment by a-saitoh-fj** [comment]
@adamjstewart 
I'm sorry, I misunderstood the meaning of 'all'.

py-dask-glm
Set depends_on to type=build in the OSS to be built when building the document.

File: var/spack/repos/builtin/packages/py-dask-glm/package.py [hunk 3]
```diff
@@ -15,10 +15,33 @@ class PyDaskGlm(PythonPackage):
...
+
     depends_on('py-setuptools', type='build')
     depends_on('py-setuptools-scm', type='build')
     depends_on('py-cloudpickle@0.2.2:', type=('build', 'run'))
     depends_on('py-dask+array', type=('build', 'run'))
     depends_on('py-multipledispatch@0.4.9:', type=('build', 'run'))
     depends_on('py-scipy@0.18.1:', type=('build', 'run'))
-    depends_on('py-scikit-learn@0.18:', type=('build', 'run'))
+    depends_on('py-scikit-learn@0.18:', type=('build', 'run'), when='~docs')
+    depends_on('py-scikit-learn@0.18:0.21.999', type='build', when='+docs')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-dask-glm/package.py]:
```suggestion
    depends_on('py-scikit-learn@0.18:0.21.999', type=('build', 'run'), when='+docs')
```
Even if docs are built, sklearn is still needed at run-time.

**Comment by a-saitoh-fj** [comment]
@adamjstewart 
What you pointed out is correct.
I fixed it.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**